apps/predictions/lib/stream_supervisor.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Predictions.StreamSupervisor do
+  @moduledoc """
+  DynamicSupervisor managing per-route streams of predictions from the API.
+  """
+
+  use DynamicSupervisor
+
+  alias Routes.Route
+
+  @spec start_link(keyword()) :: Supervisor.on_start()
+  def start_link(opts), do: DynamicSupervisor.start_link(__MODULE__, opts, name: __MODULE__)
+
+  @impl DynamicSupervisor
+  def init(_) do
+    DynamicSupervisor.init(strategy: :one_for_one)
+  end
+
+  @spec ensure_stream_is_started(Route.id_t()) :: {:ok, pid()} | :bypassed
+  def ensure_stream_is_started(route_id),
+    do: ensure_stream_is_started(route_id, System.get_env("USE_SERVER_SENT_EVENTS"))
```

Comment:
**suggestion (non-blocking):** I'm not sure exactly how dotcom is built and deployed and how releases factor into it (or don't), but if you want to be more Elixir-y about this it may be possible to [set the application configuration at runtime](https://hexdocs.pm/mix/1.14/Mix.Tasks.Release.html#module-runtime-configuration) based on the environment variable rather than querying it directly here.

Reply:
Yes!! Not ready to tackle this yet.